Example/Harmony/Core/Domain/Interactors/GetAllItemsInteractor.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  GetAllItemsInteractor.swift
+//  Harmony_Example
+//
+//  Created by Joan Martin on 22/3/21.
+//  Copyright © 2021 CocoaPods. All rights reserved.
+//
+
+import Harmony
+
+struct GetAllItemsInteractor {
+    let executor: Executor
+    let getItems: Interactor.GetAllByQuery<Item>
+    
+    func execute(_ operation: Harmony.Operation = MainOperation()) -> Future<[Item]> {
```

Comment:
We usually provide an executor (default nil) in the execute function. if an executor is set, then we run the submit using it.

let exec = executor ?? self.executor
exec.submit...

Reply:
Good point but it's mostly a good practice but not needed all the time. If you think, the interaction will be used in more interaction is good to have it already prepare it but if you don't do it, you can always do it later.

Reply:
Exactly. This is a good practice when your interactions can be reused in other interactions. But in this case, this is a "final" interaction, aka. it's not required.